### PR TITLE
feat: AutoSuggestBox porting

### LINKED ISSUE #24318: Port WinUI 1.7 AutoSuggestBox behavior to Uno Skia
### What would you like to be added? 💡

Port `AutoSuggestBox`, its event arguments, automation peer, helper, reversed vector, templates, popup positioning, keyboard/focus, SIP, selection, and corner-radius behavior from the pinned WinUI implementation.

### Why is this needed? 🤔

Uno's legacy implementation diverges from WinUI across lifecycle handling, text-change reason ordering, selection, popup layout, collection forwarding, automation, and template reapplication. No existing issue tracks the complete parity effort.

### Renderer 🎨

- [x] Skia
- [ ] Native

### For which platforms 📱💻🖥️

- [x] WebAssembly
- [x] Android
- [x] iOS
- [x] Desktop (Windows)
- [x] Desktop (macOS)
- [x] Desktop (X11)
- [x] Desktop (Linux Framebuffer)
- [x] Windows App SDK

### Anything else we need to know? 💬

Implementation PR: https://github.com/unoplatform/uno/pull/22432

Primary MUX source: `winui3/release/1.7.1`, commit `5f27a786ac96ce2a81efe91310abe1eef995ec0e`.

The imported integration tests are pinned to `microsoft/microsoft-ui-xaml` commit `8463f451`.

### PR DESCRIPTION
**Pull request by MartinZikmund**
**GitHub Issue:** closes #24318

## PR Type:

- ✨ Feature
- 🐞 Bugfix

## What is the current behavior? 🤔

The legacy Uno `AutoSuggestBox` implementation diverges from WinUI in lifecycle handling, text-change reason ordering, selection, popup positioning, collection forwarding, automation, keyboard/focus behavior, SIP handling, and template reapplication.

## What is the new behavior? 🚀

Ports the WinUI 1.7.1 `AutoSuggestBox` implementation and supporting types while retaining the Uno-specific adaptations required for Skia:

- Ports the control implementation, event arguments, automation peer, helper, reversed-vector behavior, resources, and popup/keyboard logic from pinned MUX sources.
- Preserves MUX event ordering, selection, text conversion, popup positioning, SIP deferral, and corner-radius lifecycle semantics.
- Uses `ItemsControl` ownership for item-change notifications and forwards observable-vector changes correctly.
- Adds the imported MUX integration suite plus focused Uno regression coverage for lifecycle, input, template reapplication, popup placement, and corner-radius behavior.
- Keeps generated WinAppSDK API output synchronized without marking implemented Skia APIs as unimplemented.

Primary MUX source: `winui3/release/1.7.1`, commit `5f27a786ac96ce2a81efe91310abe1eef995ec0e`.

Imported integration tests are pinned to `microsoft/microsoft-ui-xaml` commit `8463f451`.

## PR Checklist ✅

- [x] 📝 Commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.
- [x] 🧪 Runtime and integration tests cover the port and its regression fixes.
- [x] 📚 No public documentation update is required; this aligns an existing WinUI API implementation.
- [x] 🖼️ Screenshot comparison was validated with no AutoSuggestBox-specific regression.
- [x] ❗ Contains **NO** breaking API changes.

## Other information ℹ️

Validation includes:

- Complete AutoSuggestBox suite on Skia Desktop: 66/66.
- Complete AutoSuggestBox suite on Skia WebAssembly: 64/64, with two expected clipboard skips.
- Full source-generator suite: 468 passed, 14 expected skips.
- Strict Desktop, WebAssembly, and Android builds.
- Native WinUI runtime-test compilation and focused parity coverage.
- Full Azure build matrix, including Android, iOS, Desktop, WebAssembly, WinAppSDK, unit tests, sync generation, and screenshot comparison.

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/Primitives/AutoSuggestBoxHelper.cs [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+
+		var revokers = GetAutoSuggestEventRevokers(autoSuggestBox);
+		if (revokers is null)
+		{
+			return;
+		}
+
+		// Only set up popup events if not already set up
+		if (revokers.PopupOpenedRevoker is null || revokers.PopupClosedRevoker is null)
+		{
```
[the remark this answers is not in the document]
Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Controls/Primitives/AutoSuggestBoxHelper.cs:106]: Fixed in `839a7ca75f`, though not quite as either suggestion proposed — worth explaining why.

The `||` is not a typo: the WinUI C++ (`controls/dev/AutoSuggestBox/AutoSuggestBoxHelper.cpp`, `OnAutoSuggestBoxLoaded`) uses the same `if (!revokers->m_popupOpenedRevoker || !revokers->m_popupClosedRevoker)` guard. It is safe there because the revokers are `winrt::auto_revoke` handles, and **assigning to an `auto_revoke` revoker revokes the previous subscription**. So re-entering the block in C++ cannot double-subscribe.

The port stores plain `IDisposable?` fields, which have no such assignment semantics — that is where the real defect is. So rather than switching to `&&` (which would diverge from WinUI *and*, in the asymmetric case, leave one side permanently unsubscribed instead of re-wiring it), I kept the `||` guard and explicitly revoked before re-wiring, which is the faithful C# equivalent of `auto_revoke`:

```csharp
// C++ assigns to winrt::auto_revoke revokers, which revoke the previous subscription
// on assignment. Plain IDisposable fields don't, so revoke explicitly before re-wiring.
revokers.PopupOpenedRevoker?.Dispose();
revokers.PopupClosedRevoker?.Dispose();
```

Net effect: no duplicate handlers and no stale subscription, with the WinUI structure preserved.

_Reviewed by inspection only — no local build was run; Azure Pipelines CI is the validation gate._

File: src/Uno.UI/UI/Xaml/Controls/Primitives/AutoSuggestBoxHelper.cs [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+		}
+
+		var revokers = GetAutoSuggestEventRevokers(autoSuggestBox);
+		if (revokers is null)
+		{
+			return;
+		}
+
+		// Only set up popup events if not already set up
+		if (revokers.PopupOpenedRevoker is null || revokers.PopupClosedRevoker is null)
```
[the remark this answers is not in the document]
Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Controls/Primitives/AutoSuggestBoxHelper.cs:105]: Fixed in `839a7ca75f`, though not quite as either suggestion proposed — worth explaining why.

The `||` is not a typo: the WinUI C++ (`controls/dev/AutoSuggestBox/AutoSuggestBoxHelper.cpp`, `OnAutoSuggestBoxLoaded`) uses the same `if (!revokers->m_popupOpenedRevoker || !revokers->m_popupClosedRevoker)` guard. It is safe there because the revokers are `winrt::auto_revoke` handles, and **assigning to an `auto_revoke` revoker revokes the previous subscription**. So re-entering the block in C++ cannot double-subscribe.

The port stores plain `IDisposable?` fields, which have no such assignment semantics — that is where the real defect is. So rather than switching to `&&` (which would diverge from WinUI *and*, in the asymmetric case, leave one side permanently unsubscribed instead of re-wiring it), I kept the `||` guard and explicitly revoked before re-wiring, which is the faithful C# equivalent of `auto_revoke`:

```csharp
// C++ assigns to winrt::auto_revoke revokers, which revoke the previous subscription
// on assignment. Plain IDisposable fields don't, so revoke explicitly before re-wiring.
revokers.PopupOpenedRevoker?.Dispose();
revokers.PopupClosedRevoker?.Dispose();
```

Net effect: no duplicate handlers and no stale subscription, with the WinUI structure preserved.

_Reviewed by inspection only — no local build was run; Azure Pipelines CI is the validation gate._

File: src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.mux.cs [hunk 3]
```diff
@@ -0,0 +1,2147 @@
...
+	/// Gets the plain text representation of the control for accessibility.
+	/// </summary>
+	internal new string GetPlainText()
+	{
+		var header = Header;
+		if (header is not null)
+		{
+			return header.ToString();
+		}
+		return null;
```
[the remark this answers is not in the document]
Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.mux.cs]: Agreed and fixed in `839a7ca75f` — `internal new string GetPlainText()` → `internal override string GetPlainText()`.

Confirmed against both sides:
- Base: `FrameworkElement.mux.cs:13` declares `internal virtual string GetPlainText() => "";`, and `GetStringFromObject` dispatches through that virtual — so `new` meant the header text never surfaced.
- WinUI: `AutoSuggestBox_Partial.h:76` declares `GetPlainText(_Out_ HSTRING*) override`, so `override` is also the 1:1-correct port.

Every sibling control (`ContentControl`, `Slider`, `TextBox`, `MenuFlyoutItem`, …) already uses `override`, so this was simply a slip.

_Reviewed by inspection only — no local build was run; CI will verify._

File: src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.mux.cs [hunk 4]
```diff
@@ -0,0 +1,2147 @@
...
+
+			// TODO UNO: Raise automation property changed event for ControlledPeers
+			// This requires AutomationProperty.LookupById which is not available in Uno
+		}
+	}
+
+	/// <summary>
+	/// Gets the plain text representation of the control for accessibility.
+	/// </summary>
+	internal new string GetPlainText()
```
[the remark this answers is not in the document]
Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.mux.cs]: Agreed and fixed in `839a7ca75f` — `internal new string GetPlainText()` → `internal override string GetPlainText()`.

Confirmed against both sides:
- Base: `FrameworkElement.mux.cs:13` declares `internal virtual string GetPlainText() => "";`, and `GetStringFromObject` dispatches through that virtual — so `new` meant the header text never surfaced.
- WinUI: `AutoSuggestBox_Partial.h:76` declares `GetPlainText(_Out_ HSTRING*) override`, so `override` is also the 1:1-correct port.

Every sibling control (`ContentControl`, `Slider`, `TextBox`, `MenuFlyoutItem`, …) already uses `override`, so this was simply a slip.

_Reviewed by inspection only — no local build was run; CI will verify._

File: src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.mux.cs [hunk 5]
```diff
@@ -0,0 +1,2147 @@
...
+		{
+			return;
+		}
+
+		// TODO UNO: Implement LightDismissOverlayHelper
+		// bool isOverlayVisible = LightDismissOverlayHelper.ResolveIsOverlayVisibleForControl(this);
+
+		bool isOverlayVisible = false;
+		isOverlayVisible &= IsSuggestionListOpen; // Overlay should only be visible when the suggestion list is.
+		isOverlayVisible &= !s_sipIsOpen;          // Except if the SIP is also visible.
```
[the remark this answers is not in the document]
Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.mux.cs]: Correct — the `&=` operands were dead. Fixed in `839a7ca75f`, taking your suggestion.

For context on *why* it looked like that: WinUI seeds the local from the helper —

```cpp
bool isOverlayVisible = false;
IFC_RETURN(LightDismissOverlayHelper::ResolveIsOverlayVisibleForControl(this, &isOverlayVisible));
isOverlayVisible &= !!isSuggestionListOpen;
isOverlayVisible &= !m_sSipIsOpen;
```

`LightDismissOverlayHelper` does not exist in Uno yet, so the port dropped the seeding line and left `false` behind, stranding the two `&=` lines. The overlay is intentionally inert for now (`SetupOverlayState`/`TeardownOverlayState` are themselves empty TODO stubs), so the honest expression is `const bool isOverlayVisible = false;` with the WinUI conditions preserved in the TODO comment for whoever implements the helper. Behaviour is unchanged; only the dead code is gone.

_Reviewed by inspection only — no local build was run; CI is the validation gate._

**Comment by MartinZikmund** [comment]
## Review comments addressed — `aac8d87286`

5 unresolved threads, which collapse into **2 distinct issues** (each was re-reported across review rounds). One was a real leak and is fixed; the other is a false positive.

### Fixed (3 threads — all the same issue)

| file:line | reviewer point | what changed |
|---|---|---|
| `AutoSuggestBox.mux.cs:346` (x2), `:99` | SIP `Showing`/`Hiding` handlers are never revoked, so `InputPane` keeps the `AutoSuggestBox` alive | **Confirmed real.** `InputPane` is a `private static InputPane _instance = new()` singleton (`src/Uno.UI/UI/ViewManagement/InputPane/InputPane.cs:11`) exposing plain strong `event TypedEventHandler` fields — so the subscription roots the control from a static field for the whole process. The port header already *documented* this cleanup ("SIP event handlers are removed via SerialDisposable revokers **and the Unloaded event handler**"), but `OnUnloaded_Internal` never did it. Fixed. |

I did **not** apply the suggested diff verbatim, because dropping the revokers on `Unloaded` without re-establishing them would have permanently killed SIP handling for any re-loaded `AutoSuggestBox` (cached `NavigationView`/`Pivot`/`TabView` pages, virtualized items) — `SetupInputPaneEvents()` was only ever called from the constructor. The second bot comment concedes this point itself. What actually landed:

- SIP handlers are now wired in a new `OnLoaded_Internal` and revoked in `OnUnloaded_Internal` → subscriptions exist **only while the control is in the tree**, and a re-attached control gets them back.
- The teardown is gated on `!IsLoaded`, reusing the guard the C++ already had, so a reparent that raises a spurious `Unloaded` doesn't silently disable the SIP.
- `SetupInputPaneEvents()` is idempotent (the `SerialDisposable` assignment drops the previous subscription), so a repeated `Loaded` cannot double-subscribe.
- `m_tpInputPane` itself is still resolved eagerly in `PrepareState` (via `ResolveInputPane()`), because `OnGotFocus` reads `m_tpInputPane.OccludedRect`. Holding a reference *to* the singleton is harmless — it's the reverse edge (singleton → control) that leaked.

### No change needed (2 threads — same issue)

| file:line | the point | why the suggestion is wrong |
|---|---|---|
| `AutoSuggestBox.mux.cs:70` (x2) | `Items.VectorChanged += OnItemsVectorChanged` has no revoker → leak | **False positive.** `Items` is the control's *own* collection: `ItemsControl` declares `private ItemCollection _items = new ItemCollection();` and `public ItemCollection Items => _items;` (`ItemsControl.cs:41,252`). The reasoning quoted ("the `ItemCollection` object outlives the visual tree attachment") is true but irrelevant — nothing *external* holds that collection. The edge is `AutoSuggestBox → _items → delegate → AutoSuggestBox`, a self-contained cycle, which the .NET GC collects. Adding a revoker here would be dead code. (Contrast with the `InputPane` case above, where the root genuinely is a `static` field — that's exactly why one is a leak and this one isn't.) |

### Validation

Changes were **reviewed by code inspection only** — I did not build, run, or test them. CI (Azure Pipelines) will verify compilation and the runtime tests. The reasoning above is from reading `InputPane.cs`, `ItemsControl.cs` and the port's own call graph; the leak/no-leak distinction rests on which references are rooted by a `static`, which is verifiable from those sources.

Nothing was left needing @MartinZikmund's input.

File: src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.mux.cs [hunk 6]
```diff
@@ -0,0 +1,2175 @@
...
+		SizeChanged += OnSizeChanged;
+		m_sizeChangedEventRevoker.Disposable = Disposable.Create(() => SizeChanged -= OnSizeChanged);
+
+		// Resolve the InputPane eagerly (OnGotFocus reads its OccludedRect), but wire the SIP
+		// handlers only while loaded: InputPane is a process-wide singleton, so handlers left
+		// attached would root this control for the lifetime of the process.
+		ResolveInputPane();
+
+		// Hook up Items.VectorChanged
+		Items.VectorChanged += OnItemsVectorChanged;
```
[the remark this answers is not in the document]
Comment (morning4coffe-dev) [reply src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.mux.cs]: Resolved in the live branch: AutoSuggestBox no longer subscribes directly to `Items.VectorChanged`. It overrides `OnItemsChanged`, leaving subscription ownership with `ItemsControl`; `When_ItemsSource_Changes_Uses_Single_Items_Callback` verifies a single callback.

File: src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_AutoSuggestBox.cs [hunk 7]
```diff
@@ -1290,5 +1308,64 @@ public async Task When_Popup_Above_AutoSuggestBox_And_SuggestionsList_changes()
...
+			await WindowHelper.WaitForIdle();
+			await WindowHelper.WaitFor(() => SUT.IsSuggestionListOpen);
+
+			var textBox = (TextBox)SUT.GetTemplateChild("TextBox");
+			Assert.IsNotNull(textBox, "TextBox template child should exist");
+
+			// When popup is open, interior corners should be squared off
+			// The flyout might open up or down, so we check for either configuration
+			var textBoxCornerRadius = textBox.CornerRadius;
+			var isOpenDown = textBoxCornerRadius.BottomLeft == 0 && textBoxCornerRadius.BottomRight == 0;
```
[the remark this answers is not in the document]
Comment (morning4coffe-dev) [reply src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_AutoSuggestBox.cs:1356]: The corner values are assigned the literal value `0` by `CornerRadiusFilterConverter`. Exact equality is intentional here because the test verifies exact squared-corner parity; an epsilon would weaken that assertion.

[the remark this answers is not in the document]
Comment (morning4coffe-dev) [reply src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_AutoSuggestBox.cs:1356]: The corner values are assigned the literal value `0` by `CornerRadiusFilterConverter`. Exact equality is intentional here because the test verifies exact squared-corner parity; an epsilon would weaken that assertion.

File: src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_AutoSuggestBox.cs [hunk 8]
```diff
@@ -1290,5 +1308,64 @@ public async Task When_Popup_Above_AutoSuggestBox_And_SuggestionsList_changes()
...
+			var isOpenUp = textBoxCornerRadius.TopLeft == 0 && textBoxCornerRadius.TopRight == 0;
```
[the remark this answers is not in the document]
Comment (morning4coffe-dev) [reply src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_AutoSuggestBox.cs:1357]: The corner values are assigned the literal value `0` by `CornerRadiusFilterConverter`. Exact equality is intentional here because the test verifies exact squared-corner parity; an epsilon would weaken that assertion.

[the remark this answers is not in the document]
Comment (morning4coffe-dev) [reply src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_AutoSuggestBox.cs:1357]: The corner values are assigned the literal value `0` by `CornerRadiusFilterConverter`. Exact equality is intentional here because the test verifies exact squared-corner parity; an epsilon would weaken that assertion.

File: src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_AutoSuggestBox.cs [hunk 9]
```diff
@@ -1290,5 +1308,64 @@ public async Task When_Popup_Above_AutoSuggestBox_And_SuggestionsList_changes()
...
+
+			var popup = popups.Last();
+			var popupBorder = popup.Child as Border;
+			if (popupBorder != null)
+			{
+				var popupCornerRadius = popupBorder.CornerRadius;
+				// Popup corners opposite to textbox should be squared
+				if (isOpenDown)
+				{
+					Assert.IsTrue(popupCornerRadius.TopLeft == 0 && popupCornerRadius.TopRight == 0,
```
[the remark this answers is not in the document]
Comment (morning4coffe-dev) [reply src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_AutoSuggestBox.cs:1374]: The corner values are assigned the literal value `0` by `CornerRadiusFilterConverter`. Exact equality is intentional here because the test verifies exact squared-corner parity; an epsilon would weaken that assertion.

[the remark this answers is not in the document]
Comment (morning4coffe-dev) [reply src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_AutoSuggestBox.cs:1374]: The corner values are assigned the literal value `0` by `CornerRadiusFilterConverter`. Exact equality is intentional here because the test verifies exact squared-corner parity; an epsilon would weaken that assertion.

File: src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_AutoSuggestBox.cs [hunk 10]
```diff
@@ -1290,5 +1308,64 @@ public async Task When_Popup_Above_AutoSuggestBox_And_SuggestionsList_changes()
...
+						$"Popup border should have squared top corners when opening down. Actual: TL={popupCornerRadius.TopLeft}, TR={popupCornerRadius.TopRight}");
+				}
+				else
+				{
+					Assert.IsTrue(popupCornerRadius.BottomLeft == 0 && popupCornerRadius.BottomRight == 0,
```
[the remark this answers is not in the document]
Comment (morning4coffe-dev) [reply src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_AutoSuggestBox.cs:1379]: The corner values are assigned the literal value `0` by `CornerRadiusFilterConverter`. Exact equality is intentional here because the test verifies exact squared-corner parity; an epsilon would weaken that assertion.

[the remark this answers is not in the document]
Comment (morning4coffe-dev) [reply src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_AutoSuggestBox.cs:1379]: The corner values are assigned the literal value `0` by `CornerRadiusFilterConverter`. Exact equality is intentional here because the test verifies exact squared-corner parity; an epsilon would weaken that assertion.

File: src/Uno.UI/UI/Xaml/Controls/Primitives/AutoSuggestBoxHelper.cs [hunk 11]
```diff
@@ -0,0 +1,228 @@
...
+			try
+			{
+				var transform = popupBorder.TransformToVisual(textBox);
+				var popupTop = transform.TransformPoint(new global::Windows.Foundation.Point(0, 0));
+				verticalOffset = popupTop.Y;
+			}
+			catch
+			{
+				// Transform may fail if elements are not in the same visual tree
+			}
```
[the remark this answers is not in the document]
Comment (morning4coffe-dev) [reply src/Uno.UI/UI/Xaml/Controls/Primitives/AutoSuggestBoxHelper.cs]: Fixed in `637e962086`: the transform catch was removed, matching the pinned MUX direct `TransformToVisual` path. Corner and template-reapplication stress coverage passes.

File: src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.mux.cs [hunk 12]
```diff
@@ -0,0 +1,2168 @@
...
+	}
+
+	private void ResolveInputPane()
+	{
+		try
+		{
+			m_tpInputPane = InputPane.GetForCurrentView();
+		}
+		catch
+		{
```
[the remark this answers is not in the document]
Comment (morning4coffe-dev) [reply src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.mux.cs]: Fixed in `637e962086`: Uno now resolves its deterministic `InputPane` singleton directly, so unexpected failures propagate instead of being swallowed, matching MUX HRESULT behavior.

File: src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.mux.cs [hunk 13]
```diff
@@ -0,0 +1,2168 @@
...
+				m_availableSuggestionHeight = deltaTop;
+			}
+
+			// Set availableHeight to zero if there is no space left on the screen
+			if (m_availableSuggestionHeight < 0)
+			{
+				m_availableSuggestionHeight = 0;
+			}
+		}
+		catch
```
[the remark this answers is not in the document]
Comment (morning4coffe-dev) [reply src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.mux.cs]: Fixed in `637e962086`: positioning cleanup now uses `try/finally`, preserving the MUX scope-exit negative-height clamp without masking calculation errors.

File: src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.mux.cs [hunk 14]
```diff
@@ -0,0 +1,2168 @@
...
+			// Update query button's AutomationProperties.Name to "Search" by default
+			// C++ line ~843: DXamlCore::GetLocalizedResourceString(UIA_AUTOSUGGESTBOX_QUERY)
+			var automationName = AutomationProperties.GetName(m_tpTextBoxQueryButtonPart);
+			if (string.IsNullOrEmpty(automationName))
+			{
+				string searchName;
+				try
+				{
+					searchName = ResourceAccessor.GetLocalizedStringResource(ResourceAccessor.SR_AutoSuggestBoxQueryButton);
+				}
```
[the remark this answers is not in the document]
Comment (morning4coffe-dev) [reply src/Uno.UI/UI/Xaml/Controls/AutoSuggestBox/AutoSuggestBox.mux.cs]: Fixed in `637e962086`: missing or empty localized text explicitly falls back to `"Search"`; unrelated exceptions now propagate.

File: src/Uno.UI/UI/Xaml/Controls/Primitives/AutoSuggestBoxHelper.cs [hunk 15]
```diff
@@ -0,0 +1,228 @@
...
+		}
+
+		var revokers = GetAutoSuggestEventRevokers(autoSuggestBox);
+		if (revokers is null)
+		{
+			return;
+		}
+
+		// Only set up popup events if not already set up
+		if (revokers.PopupOpenedRevoker is null || revokers.PopupClosedRevoker is null)
```
[the remark this answers is not in the document]
Comment (morning4coffe-dev) [reply src/Uno.UI/UI/Xaml/Controls/Primitives/AutoSuggestBoxHelper.cs]: Already fixed in the live branch: every load/template setup disposes and replaces popup opened, closed, and border-loaded revokers. Template-reapplication coverage passes.

File: src/Uno.UI/UI/Xaml/Controls/Primitives/AutoSuggestBoxHelper.cs [hunk 16]
```diff
@@ -0,0 +1,228 @@
...
+		if (popupBorder is not null && textBox is not null)
+		{
+			try
+			{
+				var transform = popupBorder.TransformToVisual(textBox);
+				var popupTop = transform.TransformPoint(new global::Windows.Foundation.Point(0, 0));
+				verticalOffset = popupTop.Y;
+			}
+			catch
+			{
```
[the remark this answers is not in the document]
Comment (morning4coffe-dev) [reply src/Uno.UI/UI/Xaml/Controls/Primitives/AutoSuggestBoxHelper.cs]: Fixed in `637e962086`: the transform catch was removed, matching the pinned MUX direct `TransformToVisual` path. Corner and template-reapplication stress coverage passes.

File: src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_AutoSuggestBox.cs [hunk 17]
```diff
@@ -1273,5 +1301,381 @@ public async Task When_Popup_Above_AutoSuggestBox_And_SuggestionsList_changes()
...
+			}
+
+			async Task RunScenario(bool moveForward)
+			{
+				autoSuggestBox.Text = "";
+				autoSuggestBox.Focus(FocusState.Keyboard);
+				await WindowHelper.WaitForIdle();
+
+				await PressKey(VirtualKey.S, withShift: true);
+				await Task.Delay(500);
```
[the remark this answers is not in the document]
Comment (morning4coffe-dev) [reply src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_AutoSuggestBox.cs]: Fixed in `76026e6954`: the fixed 500 ms delay was replaced by a condition wait for the exact typed text, exposing input failures without timing brittleness.

File: src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_AutoSuggestBox.cs [hunk 18]
```diff
@@ -1273,5 +1301,381 @@ public async Task When_Popup_Above_AutoSuggestBox_And_SuggestionsList_changes()
...
+					await KeyboardHelper.Tab();
+				}
+#endif
+			}
+		}
+
+#if HAS_UNO
+		private static void AssertHasSquaredInteriorCorners(CornerRadius radius)
+		{
+			var opensDown = radius.BottomLeft == 0 && radius.BottomRight == 0;
```
[the remark this answers is not in the document]
Comment (morning4coffe-dev) [reply src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_AutoSuggestBox.cs]: The corner values are assigned the literal value `0` by `CornerRadiusFilterConverter`. Exact equality is intentional here because the test verifies exact squared-corner parity; an epsilon would weaken that assertion.
